docs/architecture/architecture.md

```diff
@@ -0,0 +1,671 @@
+reaper and `cc-proxy`. A `cc-shim` instance will both forward signals and `stdin`
+streams to the container process on the guest and pass the container `stdout`
+and `stderr` streams back up the stack to CRI-O or Docker via the container process
+reaper. `cc-runtime` creates a `cc-shim` daemon for each container and for each
+OCI command received to run within an already running container (example, `docker
+exec`).
+
+The container workload, that is, the actual OCI bundle rootfs, is exported from the
+host to the virtual machine.  In the case where a block-based graph driver is
+configured, virtio-blk will be used. In all other cases a 9pfs virtio mount point
+will be used. `cc-agent` uses this mount point as the root filesystem for the
+container processes.
+
+![Overall architecture](arch-images/overall-architecture.png)
+
+## Hypervisor
+
+Clear Containers use  [QEMU](http://www.qemu-project.org/)/[KVM](http://www.linux-kvm.org/page/Main_Page)
+to create virtual machines where containers will run:
+
+![QEMU/KVM](arch-images/qemu.png)
+
+Although Clear Containers can run with any recent QEMU release, Clear Containers
+boot time and memory footprint are significantly optimized by using a specific QEMU
+version called [`qemu-lite`](https://github.com/01org/qemu-lite/tree/qemu-2.7-lite).
+
+Clear Containers supports various machine types, including `pc`, `pc-lite` and `q35`.
+Clear Containers defaults to using the `pc` machine type. In the past
+`pc-lite` was utilized which provided the following improvements:
+- Removed many of the legacy hardware devices support so that the guest kernel
+  does not waste time initializing devices of no use for containers.
+- Skipped the guest BIOS/firmware and jumped straight to the Clear Containers
+  kernel.
+
+In order to provide some advanced features which were lacking in `pc-lite` like hotplug
+support, the default was changed to `pc`.
+
+In the future, Clear Containers plan to move to a 2.9 based version of QEMU,
+available at 
+the [Clear Containers QEMU repo](https://github.com/clearcontainers/qemu/tree/qemu-lite-v2.9.0).
+This transition has been delayed until after the release of Clear Containers 3.0
+due to regressions, as described in [runtime issue 407](https://github.com/clearcontainers/runtime/issues/407).
+Once support for features like hotplug are available in `Q35`, the project will
+look to transition to this machine type.
+ 
+## Agent
+
+[`cc-agent`](https://github.com/clearcontainers/agent) is a daemon running in the
+guest as a supervisor for managing containers and processes potentially running
+within those containers.
+
+The `cc-agent` execution unit is the pod. A `cc-agent` pod is a container sandbox
+defined by a set of namespaces (NS, UTS, IPC and PID). `cc-runtime` can run several
+containers per pod to support container engines that require multiple containers
+running inside a single VM. In the case of docker, `cc-runtime` creates a single
```

Comment:
We haven't defined what "VM" means. I suggest we add a Terminology section for this and other such terms.